From d54567585f7cc475fb1298f53d05f53d653dbb20 Mon Sep 17 00:00:00 2001
From: Nitin Garg <nitin.garg_3@nxp.com>
Date: Tue, 12 Nov 2024 12:42:32 -0600
Subject: [PATCH] YOCIMX-8508 net: imx-shmem-net: Fix
 -Wincompatible-pointer-types

Fixes build break:
```
drivers/net/imx-shmem-net.c:1259:27: error: initialization of 'void (*)(struct platform_device *)' from incompatible pointer type 'int (*)(struct platform_device *)' [-Wincompatible-pointer-types]
1259 |         .remove         = imx_shm_net_remove,
     |                           ^~~~~~~~~~~~~~~~~~
```

Upstream-Status: Pending
Signed-off-by: Nitin Garg <nitin.garg_3@nxp.com>
Signed-off-by: Tom Hochstein <tom.hochstein@nxp.com>
---
 drivers/net/imx-shmem-net.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/imx-shmem-net.c b/drivers/net/imx-shmem-net.c
index 4db888c7c597..2556636a1230 100644
--- a/drivers/net/imx-shmem-net.c
+++ b/drivers/net/imx-shmem-net.c
@@ -1188,7 +1188,7 @@ static int imx_shm_net_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int imx_shm_net_remove(struct platform_device *pdev)
+static void imx_shm_net_remove(struct platform_device *pdev)
 {
 	struct net_device *ndev = platform_get_drvdata(pdev);
 	struct imx_shm_net *in = netdev_priv(ndev);
@@ -1217,8 +1217,6 @@ static int imx_shm_net_remove(struct platform_device *pdev)
 	cancel_work_sync(&in->state_work);
 	destroy_workqueue(in->state_wq);
 	free_netdev(ndev);
-
-	return 0;
 }
 
 #ifdef CONFIG_PM_SLEEP
-- 
2.34.1

